### ISSUE TITLE
[v2] Member is not recognized as a Snowflake by PyCharm's type checker.

### ISSUE THREAD

**Issue by Iterniam** [Rapptz/discord.py#7291]
### Summary

[v2] Member is not recognized as a Snowflake by PyCharm's type checker.

### Reproduction Steps

I fed a method that expects `list[Snowflake]` something of type `list[Optional[Member]]`.

### Minimal Reproducible Code

```python
@commands.command('test')
async def test(self, ctx: Context):
    bot: Bot = ctx.bot  # Else ctx.bot is not recognized as Bot
    guild: Guild = ctx.guild  # Else ctx.guild is not recognized as Guild
    user = bot.get_user(191197956769054721)
    member = guild.get_member(191197956769054721)
    _ = AllowedMentions(users=[user])  # Passes the type checker
    _ = AllowedMentions(users=[member])  # Type checker complains
```

### Expected Results

I expect an instance of `Member` to be recognized as a valid `Snowflake`, similar to how `User` is recognized as a `Snowflake`.

### Actual Results

I get the following warning from the type checker: `Expected type 'Union[bool, list[Snowflake]]', got 'list[Optional[Member]]' instead`

### Intents

Intents.default()

### System Information

The command returned an empty line, but the version of discord.py is `2.0.0a3407+gfc51736b`

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

This bug is not present in the current stable version of discord.py (1.7.3)

**Comment by Phxntxm** [comment]
Neither of those should pass (both fail as they should in VSC for example), if Pycharm has that `user` one passing then that is a mistake by Pycharm. The type expected is `List[Snowflake]` not `List[Optional[Snowflake]]`.

**Comment by Iterniam** [comment]
Pycharm's type checker generally doesn't complain if something is `Optional[T]` instead of `T`.

That said, here's a more robust example that still fails makes the type checker fail:

```py

def returns_user() -> User:
    ...
def returns_member() -> Member:
    ...

class Test(Cog):
    @commands.command('test')
    async def test(self, ctx: Context):
        _ = AllowedMentions(users=[returns_user()])
        _ = AllowedMentions(users=[returns_member()])
```

The actual result is now `Expected type 'Union[bool, list[Snowflake]]', got 'list[Member]' instead`

**Comment by mrvillage** [comment]
In 2.0, discord.Member does not inherit from Snowflake, don't know if that was an intended change or not.

**Comment by SebbyLaw** [comment]
> In 2.0, discord.Member does not inherit from Snowflake, don't know if that was an intended change or not.

`discord.Member` has never inherited from Snowflake.

I can reproduce this issue in pycharm on commits before d09993d7. Seems like this issue was fixed in #7442.
